packages/teams-js/src/internal/communication.ts

```diff
@@ -611,42 +621,52 @@ const handleParentMessageLogger = communicationLogger.extend('handleParentMessag
 function handleParentMessage(evt: DOMMessageEvent): void {
   const logger = handleParentMessageLogger;
 
-  if ('id' in evt.data && typeof evt.data.id === 'number') {
+  if ('id' in evt.data && (typeof evt.data.id === 'number' || typeof evt.data.id === 'string')) {
     // Call any associated Communication.callbacks
     const message = evt.data as MessageResponse;
-    const callback = CommunicationPrivate.callbacks[message.id];
-    logger('Received a response from parent for message %i', message.id);
+    const callbackId =
+      typeof message.id === 'string' ? message.id : CommunicationPrivate.legacyMessageIdsToUuidMap[message.id];
```

Comment:
>typeof message.id === 'string' [](http://example.com/codeflow?start=6&length=30)

I think you are trying to determine whether the id is a UUID or a number here. If this code "knows" that a UUID is just a string format, then there really isn't any value in using the custom type aliases you defined in `messageObjects.ts` (`MessageUUID` and `BaseUUID`) since other parts of the code are ignoring that alias and just going to the underlying data.